git-info-generator.mjs

```diff
@@ -0,0 +1,10 @@
+import  shell from 'shelljs';
+
+if (!shell.which('git')) {
+  throw new Error(`"git is missing", please install git and retry`);
+}
+shell
+  .exec(
+    `echo \"$(git rev-parse --abbrev-ref HEAD)\n$(git log -1 --pretty=format:%h)\n$(git describe --tags --abbrev=0)\"`
```

Comment:
Any reason to be running shell code like this as opposed to using [`child_process.spawnSync(...)`](https://nodejs.org/docs/latest-v16.x/api/child_process.html#child_processspawnsynccommand-args-options)?

Reply:
It is done this way to support it on all platform including windows. Otherwise OS there would be different code for windows like [spawning-bat-and-cmd-files-on-windows](https://nodejs.org/api/child_process.html#spawning-bat-and-cmd-files-on-windows)

Reply:
```ts
const gitInfo = [
  'git rev-parse --abbrev-ref HEAD',
  'git log -1 --pretty=format:%h',
  'git describe --tags --abbrev=0',
].map(r => execSync(r, { encoding: 'utf8' }));
fs.writeFileSync('.git-info', gitInfo.join('\n'));
```
I think something like this would work just fine

Reply:
@zone117x What if the user has git installed but does not have tags fetch locally. should we throw an error in that case or ignore it?

Reply:
Good question, I think if git repo data is missing from disk then this should throw. @rafaelcr thoughts?

Reply:
Yes, that's a very good point @asimm241. IIRC, though, git doesn't return status 1 if the tags are missing from the repo so you'll probably have to check the output message directly.